### PR TITLE
Bug 1872996 - Detekt should not warn of LongParameterList for Compose functions

### PR DESCRIPTION
**Pull request by t-p-white**
Would appreciate thoughts around the approach for this. We can either:

1. turn off the rule for Compose functions completely
2. increase the threshold for all functions (suggest we start with 8 & see how that works?)

[Detekt guide for handling Compose](https://detekt.dev/docs/introduction/compose/#longparameterlist-for-compose).

Subsequent bug should remove all cases where Compose annotations have this suppression.

With the help of some hacky regex, I've analysed the total number of parameters for `Composable` funcs with 10 or more parameters (note not all of these require suppressions as they have default values set):

10 params: BrowserToolbar, TopSites, ImageLoader
11 params: TabGridItem, TabListItem, TabsTrayPreviewRoot, OnboardingContent, 
12 params: OnboardingScreen
13 params: ProductAnalysisError, TabLayouty NoAnalysis, TabGrid, TabList 
14 params: ProductReview
17 params: ProductAnalysis
18 params: TabsTrayBanner, NormalTabsPage
36 params: TabsTray

Hacky regex for reference 
```@Composable\n.*?fun \w*\(\n().*?,```
```@Composable\n@Suppress\(.*?\)\n.*?fun \w*\(\n().*?,```

### Pull Request checklist

- [ ] **Quality**: This PR builds and passes detekt/ktlint checks (A pre-push hook is recommended)
- [ ] **Tests**: This PR includes thorough tests or an explanation of why it does not
- [ ] **Changelog**: This PR includes [a changelog entry](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/changelog.md) or does not need one
- [ ] **Accessibility**: The code in this PR follows [accessibility best practices](https://github.com/mozilla-mobile/firefox-android/blob/main/docs/shared/android/accessibility_guide.md) or does not include any user facing features

### After merge
- **Breaking Changes**: If this is a breaking Android Components change, please push a draft PR on [Reference Browser](https://github.com/mozilla-mobile/reference-browser) to address the breaking issues.

### To download an APK when reviewing a PR (after all CI tasks finished running):
1. Click on `Checks` at the top of the PR page.
2. Click on the `firefoxci-taskcluster` group on the left to expand all tasks.
3. Click on the `build-apk-{fenix,focus,klar}-debug` task you're interested in.
4. Click on `View task in Taskcluster` in the new `DETAILS` section.
5. The APK links should be on the right side of the screen, named for each CPU architecture.

### GitHub Automation
https://bugzilla.mozilla.org/show_bug.cgi?id=1872996

### CODE REVIEW

File: fenix/config/detekt.yml [hunk 1]
```diff
@@ -139,6 +139,7 @@ complexity:
     ignoreDefaultParameters: true # Enabled in https://github.com/mozilla-mobile/android-components/issues/10835
     ignoreDataClasses: true
     ignoreAnnotatedParameter: []
+    ignoreAnnotated: ['Composable']
```
Comment (gabrielluong) [inline fenix/config/detekt.yml:142]:
```suggestion
    ignoreAnnotated: ['Composable'] # Configuration for Compose https://bugzilla.mozilla.org/show_bug.cgi?id=1872996
```
We just want to reference non-default changes with a comment of where the changes came from. This helps us when we need to bump detekt version and update our detekt.yml file with the base configuration changes which we pull from https://github.com/detekt/detekt/blob/main/detekt-core/src/main/resources/default-detekt-config.yml.

**Comment by gabrielluong** [comment]
I think the current changes to ignore the Composable annotation is fine for now. I think the options will require some more thoughts because we do want to have a general understanding of how big our parameter list is and not just set it to the maximum.

**Comment by gabrielluong** [comment]
Could you also delete and regenerate the baselines? You will also need to make the changes to all the detekt files in Focus and AC.

**Comment by t-p-white** [comment]
Thanks for the feedback @gabrielluong, much appreciated!

**Comment by gabrielluong** [comment]
> Thanks for the feedback @gabrielluong, much appreciated! I'm wondering if we should start by increasing the threshold to 8 and see how that works for us?

Given your analysis:

```
10 params: BrowserToolbar, TopSites, ImageLoader
11 params: TabGridItem, TabListItem, TabsTrayPreviewRoot, OnboardingContent,
12 params: OnboardingScreen
13 params: ProductAnalysisError, TabLayouty NoAnalysis, TabGrid, TabList
14 params: ProductReview
17 params: ProductAnalysis
18 params: TabsTrayBanner, NormalTabsPage
36 params: TabsTray
```

I am wondering why increasing 8 would be the preferred choice here instead of just adding `ignoreAnnotated: ['Composable']`?

**Comment by gabrielluong** [comment]
I think bumping `functionThreshold` to a higher value is fine. Were we also planning to set `ignoreDefaultParameters = true` as well?

**Comment by t-p-white** [comment]
@gabrielluong awesome 🙂 

> Were we also planning to set ignoreDefaultParameters = true as well?

afaict we are already on L:139 right?

File: android-components/README.md [hunk 2]
```diff
@@ -290,7 +290,7 @@ If your build fails, you may find you get more instructive error messages by att
 # Coding Standards #
 
 ## Style ##
-We follow the style enforced by [ktlint](https://ktlint.github.io/) and [detekt](https://arturbosch.github.io/detekt/). See [how to configure Android Studio appropriately](https://github.com/pinterest/ktlint#option-1-recommended).
```
Comment (t-p-white) [inline android-components/README.md:293]: https://arturbosch.github.io/detekt/ is not a valid URL anymore

**Comment by gabrielluong** [comment]
> @gabrielluong awesome 🙂
> 
> > Were we also planning to set ignoreDefaultParameters = true as well?
> 
> afaict we are already on L:139 right?

Ah, that is true. Thanks!

**Review by gabrielluong — APPROVED** [review]
Changes look good. I am wondering if we were able to remove any of the suppress `LongParameterList` now along with these changes?

**Comment by t-p-white** [comment]
@gabrielluong thanks! Sure thing, I've created a [new bug](https://bugzilla.mozilla.org/show_bug.cgi?id=1874861) for this.

**Pull request merged**